### PR TITLE
logger + log level updates

### PR DESCRIPTION
**Pull request by jlewitt1**
Set log level on cluster according to local `RH_LOG_LEVEL` env var. If not set use the default

### CODE REVIEW

File: docs/debugging-logging.rst [hunk 1]
```diff
@@ -21,12 +21,16 @@ Alternatively, to see logs on your local machine while running a remote function
...
+----------
+You can set the log level to control the verbosity of the Runhouse logs. You can adjust the log level by
+setting the environment variable ``RH_LOG_LEVEL`` to your desired level.
 
 Debugging
 ~~~~~~~~~
 
 For general debugging that doesn't occur within remote function calls, you can add
-``import pdb; pdb.set_trace()`` whereever you want to set your debugging session.
+``import pdb; pdb.set_trace()`` wherever you want to set your debugging session.
```
Comment (rohinb2) [inline docs/debugging-logging.rst:33]: should replace this with just `breakpoint()`

File: runhouse/logger.py [hunk 2]
```diff
@@ -5,21 +5,21 @@
 def get_logger(name: str = __name__):
     logger = logging.getLogger(name)
 
+    # Clear any existing handlers - avoid duplicate logs and maintain consistent log setup across modules
+    logger.handlers.clear()
```
Comment (rohinb2) [inline runhouse/logger.py:9]: wait won't this clear each time each file calls `get_logger`? so when each file is imported and runs... we'll be running tons of these clear and re-adds

File: tests/fixtures/docker_cluster_fixtures.py [hunk 3]
```diff
@@ -379,6 +379,7 @@ def docker_cluster_pk_ssh(request, test_org_rns_folder):
         ],
         working_dir=None,
         name="default_env",
+        env_vars={"RH_LOG_LEVEL": os.getenv("RH_LOG_LEVEL") or TESTING_LOG_LEVEL},
```
Comment (rohinb2) [inline tests/fixtures/docker_cluster_fixtures.py:382]: oh if you set this locally will it propagate server side?

Comment (jlewitt1) [reply tests/fixtures/docker_cluster_fixtures.py:382]: yea as part of the restart flow / `env.to(cluster)`

File: runhouse/logger.py (hunk 2 above)
Comment (jlewitt1) [reply runhouse/logger.py]: yea good call we should kill that

**Review by rohinb2 — APPROVED** [review]
solid

**Comment by rohinb2** [comment]
### Merge activity

* **Aug 27, 5:50 PM EDT**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1178).

**Pull request merged**